### DISCUSSION TITLE
Text on Tags component

### DISCUSSION THREAD

**Discussion by ninavizz** [adobe/react-spectrum#7356] [General]
Hello!

On the [Tag component](https://react-spectrum.adobe.com/react-spectrum/TagGroup.html), I noticed the text is vertically off-center in the object—I'm assuming because it uses a standard `body` text token, and not a token for UI elements w/o the added line height. Two questions:

1. Are there any plans to fix the vertical centering of the text in the React Spectrum tag component?
2. Will the typesetting & tokens be rethought at all for Spectrum2, such that content and UI elements don't necessarily need to share all of the same CSS attributes?

Thx!
:) nina

**Comment by snowystinger** [comment]
There were a lot of issues with the AdobeClean font and vertical centering in S1 with manual padding to center it. It should be much easier in S2.
I am assuming you are not using AdobeClean and that is why you're seeing it as off center?

**Reply by ninavizz** [reply]
Yupper! Any idea when S2 may be publicly available? 

Thx for the quick reply!!

**Reply by ninavizz** [reply]
Oooh! Sorry to be a pest @snowystinger but I just took a peek at the S2 tags in its Storybook, and they are so lovely! Might they be supporting icons as the S1's do? No biggie if not. Just curious. Thx again!

**Reply by snowystinger** [reply]
Yep, if you scroll down a bit further, you'll find an example with icons

**Reply by ninavizz** [reply]
@snowystinger I did see that, just no "controls" that implied they could be managed with basic props. TY!! :)

**Reply by snowystinger** [reply]
it's just children, same as before https://react-spectrum.adobe.com/react-spectrum/TagGroup.html#with-icons

You can create a wrapper for it and control icons however you like though
